mango/container/mp.py

```diff
@@ -444,6 +449,21 @@ def _find_sp_queue(self, aid):
         raise ValueError(f"The aid '{aid}' does not exist in any subprocess.")
 
     def create_agent_process(self, agent_creator, container, mirror_container_creator):
+        # dill must be dumped on creation, otherwise the later variable state would be stored
+        agent_creator = dill.dumps(agent_creator)
+        mirror_container_creator = dill.dumps(mirror_container_creator)
+        if not self._started:
+            self._agent_process_init_list.append(
+                (agent_creator, container, mirror_container_creator)
+            )
+        else:
+            self.create_internal_agent_process(
```

Comment:
create_internal_agent_process returns the process handle, which should be returned at this point as well. This also raises the question, how would you retrieve the pid in the delayed start?

Reply:
In which cases is the Pid of the agent_process useful? Expect for not breaking the API?

Yes, that's a reason for a separate function. Do you have a name suggestion?
`as_agent_process` would then be an async function (as it currently already returns an awaitable and requires to have a running loop)

and the new sync function would then be... something like `create_agent_process` or `as_agent_process_sync` or..?

Reply:
Regarding the name, sync could be misleading here as this method behaves differently. Maybe, as_agent_process_lazy?